src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -903,16 +903,12 @@ public boolean hasGracefulShutdownStarted() {
                         lease, notificationCompleteLatch, shutdownCompleteLatch);
                 ShardInfo shardInfo = KinesisClientLibLeaseCoordinator.convertLeaseToAssignment(lease);
                 ShardConsumer consumer = shardInfoShardConsumerMap.get(shardInfo);
-                if (consumer != null) {
-                    consumer.notifyShutdownRequested(shutdownNotification);
-                } else {
-                    //
-                    // There is a race condition between retrieving the current assignments, and creating the
-                    // notification. If the a lease is lost in between these two points, we explicitly decrement the
-                    // notification latches to clear the shutdown.
-                    //
+
+                if (consumer == null || ConsumerStates.ShardConsumerState.SHUTDOWN_COMPLETE.equals(consumer.getCurrentState())) {
```

Comment:
Can we preserve the existing comment and a new comment for the shutdown_complete case?

Reply:
Sure, added the comment back. Also added another comment explaining the shutdown case.